client/state/data-layer/analytics/adTracking.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	trackCustomAdWordsRemarketingEvent,
+	trackCustomFacebookConversionEvent,
+} from 'lib/analytics/ad-tracking';
+import { SIGNUP_STEPS_SURVEY_SET } from 'state/action-types';
+
+function recordSelectedVertical( _, action, next ) {
+	next( action );
```

Comment:
okay so this here is the first time I've actually seen in the wild a reason _not_ to put more stuff in the data layer that isn't data related (cc: @aduth, @ockham, @coderkevin, @gwwar, @jake, @blowery, @mtias, _et. al._ for the sake of review knowledge)

`next( action )` is going to wrap this Redux action in the bypass flag for the data layer, but this middleware isn't actually providing any data; it's just firing off an action in response to some other action.

I think I may have suggested @markryall that you tack it in here but this will be wrong and could lead to errors in the code. we need to create some separate `action-handlers` middleware which borrows all of the same elements of the data layer middleware but which doesn't wrap `next()` in `local()`.

the data layer _provides_ and _updates_ data and that is why `next()` is wrapped (so we don't try and provide the same data by multiple providers) but general action-based middleware actions _shouldn't_ cause actions to skip the data layer.